### ISSUE TITLE
Java 17 UUID can no longer be serialized and deserialized correctly

### ISSUE THREAD

**Issue by h143570** [jdereg/json-io#164]
on Java 17
```
UUID uuid = UUID.randomUUID();
JsonWriter.objectToJson(uuid);
```

Produces:

`{"@type":"java.util.UUID","mostSigBits":null,"leastSigBits":null}`

Restoring the original state is not possible from this.

Java 8 and 11 works perfectly. 

Deserialization will fail on `mostSigBits`
detailMessage: `IllegalAccessException setting field 'mostSigBits' on target: 00000000-0000-0000-0000-000000000000 with value: null`
cause: `java.lang.IllegalAccessException: class com.cedarsoftware.util.io.ObjectResolver cannot access a member of class java.util.UUID (in module java.base) with modifiers "private final"`

**Comment by jdereg** [comment]
In the User Guide (https://github.com/jdereg/json-io/blob/master/user-guide.md), option 4 under customizations shows how to add a customer serializer.  This is how json-io is extended to handle objects that do not serialize in a standard way.  Essentially, come up with a simple format for serializing the GUID - which I would expect would be a simple key associated to String value.  Write the "reader" and "writer" instance to read and write that, associate it to the class that is failing, and that class will now serialize fine regardless of Java version.  In all cases, it will use your serializer for that class.

**Comment by jdereg** [comment]
Looking forward to a PR - will include it and then will release the project with the new PR.

**Comment by h143570** [comment]
I have the initial changes still working on the unit test, based on the BigDecimal test. Likely send in the PR tomorrow.

**Comment by jdereg** [comment]
These changes have been merged and released in json-io 4.14.0
